dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/ESMappingAPIImpl.java

```diff
@@ -436,262 +436,262 @@ protected void loadPermissions(Contentlet con, Map<String,String> m) throws DotD
+							List<FieldVariable> fieldVariables=APILocator.getFieldAPI().getFieldVariablesForField(
+									f.getInode(), APILocator.getUserAPI().getSystemUser(), false);
+							for(FieldVariable fv : fieldVariables) {
+								if(fv.getKey().equals(ESMappingConstants.DOT_INDEX_PATTERN)) {
+									String[] names=fv.getValue().split(",");
+									allowedFields=new HashSet<String>();
+									for(String n : names)
+										allowedFields.add(n.trim().toLowerCase());
+								}
+							}
+							// aditional fields from the configuration file
+							String configFields=Config.getStringProperty("INDEX_METADATA_FIELDS", "");
+							if(configFields.trim().length()>0) {
+								String[] names=configFields.split(",");
+								if(names.length>0 && allowedFields==null)
+									allowedFields=new HashSet<String>();
+								for(String n : names)
+									allowedFields.add(n.trim().toLowerCase());
+							}
+						}
+
+						if(keyValueMap!=null && !keyValueMap.isEmpty())
+							for(String key : keyValueMap.keySet())
+								if(allowedFields==null || allowedFields.contains(key.toLowerCase()))
+									m.put(st.getVelocityVarName() + "." + f.getVelocityVarName() + "." + key, (String)keyValueMap.get(key));
+					}
+				} else if(f.getFieldType().equals(Field.FieldType.TAG.toString())) {
+
+					StringBuilder personaTags = new StringBuilder();
+					StringBuilder tagg = new StringBuilder();
+					List<Tag> tagList = APILocator.getTagAPI().getTagsByInode(con.getInode());
+					if(tagList ==null || tagList.size()==0) continue;
+
+					final String tagDelimit = Config.getStringProperty("ES_TAG_DELIMITER_PATTERN", ",,");
+
+
+					for ( Tag t : tagList ) {
+						if(t.getTagName() ==null) continue;
+						String myTag = t.getTagName().trim();
+						tagg.append(myTag).append(tagDelimit);
+						if ( t.isPersona() ) {
+							personaTags.append(myTag).append(' ');
+						}
+					}
+					if(tagg.length() >tagDelimit.length()){
+						String taggStr = tagg.substring(0, tagg.length()-tagDelimit.length());
+						m.put(st.getVelocityVarName() + "." + f.getVelocityVarName(), taggStr.replaceAll(",,", " "));
+						m.put(ESMappingConstants.TAGS, taggStr);
+					}
+
+
+					if ( Structure.STRUCTURE_TYPE_PERSONA != con.getStructure().getStructureType() ) {
+						if ( personaTags.length() > tagDelimit.length() ) {
+							String personaStr = personaTags.substring(0, personaTags.length()-tagDelimit.length());
+							m.put(st.getVelocityVarName() + "."+ESMappingConstants.PERSONAS, personaStr);
+							m.put(ESMappingConstants.PERSONAS, personaStr);
+						}
+					}
+
+				} else {
+					if (f.getFieldContentlet().startsWith(ESMappingConstants.FIELD_ELASTIC_TYPE_BOOLEAN)) {
+						m.put(st.getVelocityVarName() + "." + f.getVelocityVarName(), valueObj.toString());
+					} else if (f.getFieldContentlet().startsWith(ESMappingConstants.FIELD_ELASTIC_TYPE_FLOAT) || f.getFieldContentlet().startsWith(ESMappingConstants.FIELD_ELASTIC_TYPE_INTEGER)) {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Split this 181 characters long line (which is greater than 150 authorized). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00103)